File: src/test/testdata/inspections/flushInsideLoop/Test2.java
package flushInsideLoop;

import com.badlogic.gdx.graphics.glutils.ShapeRenderer;
import com.badlogic.gdx.math.MathUtils;

public class Test2 {

    void test() {
        Clazz c7 = new Clazz();

        while (MathUtils.randomBoolean()) {
            <warning>c7.getI()</warning>;
            <warning>c7.bla()</warning>;
            c7.setI(3);
            <warning>c7.setJ(3)</warning>;
            c7.getJ();

            Clazz1 clazz1 = <warning>new Clazz1()</warning>;

            Clazz2 clazz2 = new Clazz2();
            Clazz2 clazz2a = <warning>new Clazz2("dfs")</warning>;
        }
        
    }

}

File: src/test/testdata/inspections/flushInsideLoop/Test2.kt
package flushInsideLoop

import com.badlogic.gdx.graphics.g2d.PolygonSpriteBatch
import com.badlogic.gdx.graphics.g2d.SpriteBatch


class Clazz {

  var i: Int = 0
    get() {
      PolygonSpriteBatch().flush()
      return 1
    }

  var j: Int = 0
    set(value) {
      PolygonSpriteBatch().flush()
      j = value
    }

  fun bla() {
    PolygonSpriteBatch().flush()
  }


}

class Clazz1 {
  init {
    SpriteBatch().transformMatrix = null
  }
}

class Clazz2 {
  constructor() {

  }

  @Suppress("UNUSED_PARAMETER")
  constructor(str: String) {
    SpriteBatch().transformMatrix = null
  }
}